File: isma-intg-lib/isma-intg-lib-rkfehlberg/src/main/java/ru/nstu/isma/intg/lib/rungeKutta/rungeKuttaFehlberg/internal/RkFehlbergStage5Calculator.java
package ru.nstu.isma.intg.lib.rungeKutta.rungeKuttaFehlberg.internal;

import ru.nstu.isma.intg.api.methods.*;

public class RkFehlbergStage5Calculator extends StageCalculator {
	@Override
	public double yk(final double step, final double y, final double f, final double[] stages) {
		return y + 2.032407407 * stages[0] - 8.0 * stages[1] + 7.1734892788 * stages[2] - 0.2058966862 * stages[3];
	}

	@Override
	public double k(final double step, final double y, final double f, final double[] stages, final double stagesY,
			final double stagesF) {
		return step * stagesF;
	}
}

File: isma-solver/lib/rkfehlberg/src/main/kotlin/ru/nstu/isma/intg/lib/rungeKutta/rungeKuttaFehlberg/internal/RkFehlbergStage5Calculator.kt
package ru.nstu.isma.intg.lib.rungeKutta.rungeKuttaFehlberg.internal

import ru.nstu.isma.intg.api.methods.StageCalculator

class RkFehlbergStage5Calculator : StageCalculator() {
    override fun yk(step: Double, y: Double, f: Double, stages: DoubleArray): Double {
        return y + 2.032407407 * stages[0] - 8.0 * stages[1] + 7.1734892788 * stages[2] - 0.2058966862 * stages[3]
    }

    override fun k(
        step: Double, y: Double, f: Double, stages: DoubleArray, stagesY: Double,
        stagesF: Double
    ): Double {
        return step * stagesF
    }
}
